src/mocks/MockERC20.sol

```diff
@@ -1,36 +1,54 @@
 // SPDX-License-Identifier: MIT
 pragma solidity >=0.6.2 <0.9.0;
 
+import {IERC20} from "../interfaces/IERC20.sol";
+
 /// @notice This is a mock contract of the ERC20 standard for testing purposes only, it SHOULD NOT be used in production.
 /// @dev Forked from: https://github.com/transmissions11/solmate/blob/0384dbaaa4fcb5715738a9254a7c0a4cb62cf458/src/tokens/ERC20.sol
-contract MockERC20 {
+contract MockERC20 is IERC20 {
     /*//////////////////////////////////////////////////////////////
-                                 EVENTS
+                            METADATA STORAGE
     //////////////////////////////////////////////////////////////*/
 
-    event Transfer(address indexed from, address indexed to, uint256 amount);
+    string private _name;
 
-    event Approval(address indexed owner, address indexed spender, uint256 amount);
+    string private _symbol;
 
-    /*//////////////////////////////////////////////////////////////
-                            METADATA STORAGE
-    //////////////////////////////////////////////////////////////*/
+    uint8 private _decimals;
 
-    string public name;
+    function name() external view override returns (string memory) {
+        return _name;
+    }
 
-    string public symbol;
+    function symbol() external view override returns (string memory) {
+        return _symbol;
+    }
 
-    uint8 public decimals;
+    function decimals() external view override returns (uint8) {
+        return _decimals;
+    }
 
     /*//////////////////////////////////////////////////////////////
                               ERC20 STORAGE
     //////////////////////////////////////////////////////////////*/
 
-    uint256 public totalSupply;
+    uint256 private _totalSupply;
+
+    mapping(address => uint256) private _balances;
 
-    mapping(address => uint256) public balanceOf;
+    mapping(address => mapping(address => uint256)) private _allowances;
```

Comment:
Nit, let's keep the names the same and just add the underscore 
```suggestion
    mapping(address => uint256) private _balanceOf;

    mapping(address => mapping(address => uint256)) private _allowance;
```

Reply:
Sg